jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/list/CaseInstanceListPresenter.java

```diff
@@ -75,10 +95,19 @@ public void createCaseInstance() {
         newCaseInstancePresenter.show();
     }
 
-    protected void refreshData() {
+    private void caseInstancesServiceCall(int currentPage) {
         caseService.call((List<CaseInstanceSummary> cases) -> {
-            view.setCaseInstanceList(cases);
-        }).getCaseInstances(view.getCaseInstanceSearchRequest());
+            visibleCaseInstances.addAll(cases);
+            view.hideLoadButton(cases.size() < PAGE_SIZE);
+            view.setCaseInstanceList(visibleCaseInstances.stream().collect(toList()));
+        }).getCaseInstances(view.getCaseInstanceSearchRequest(),
+                            currentPage,
+                            PAGE_SIZE);
+    }
+
+    protected void refreshData() {
+        visibleCaseInstances.clear();
```

Comment:
the items are added to the view as they're loaded, but the actual list itself is taken care of in the ViewImpl when we set the model in the setCaseMilestoneList.  the list of items (as they're loaded) is maintained there (in the caseMilestoneList field).  hope i am understanding your concern correctly.